src/components/ui/dialog-modal.tsx

```diff
@@ -0,0 +1,225 @@
+import * as React from "react"
+import { MdClose } from "react-icons/md"
+import { tv, type VariantProps } from "tailwind-variants"
+import * as DialogPrimitive from "@radix-ui/react-dialog"
+
+import { cn } from "@/lib/utils/cn"
+
+import { Button } from "./buttons/Button"
+import { Center, Flex } from "./flex"
+
+const dialogVariant = tv({
+  slots: {
+    content:
+      /**
+       * `size-[calc(-2rem_+_100%)]` provides 16px x- and y-spacing
+       * around the container instead of using `margin` to
+       * avoid positioning side-effects.
+       * (The container is fixed-positioned)
+       *
+       * Credit: Chakra v2 takes a similar approach for modal
+       */
+      "data-[state=open]:animate-contentShow size-[calc(-2rem_+_100%)] fixed left-1/2 top-1/2 grid -translate-x-1/2 -translate-y-1/2 gap-4 rounded-md bg-background p-8 shadow-[hsl(206_22%_7%_/_35%)_0px_10px_38px_-10px,_hsl(206_22%_7%_/_20%)_0px_10px_20px_-15px] focus:outline-none z-modal  overflow-auto",
```

Comment:
Note that I add `overflow-auto` here in conjunction to setting `height` (see above comment) to allow for scroll when the container has a lot of content.

I'm torn here. When applying this quick fix, this creates space on the top and bottom around the container to prevent the container from hitting the edge of the screen. Prod does not have bottom spacing because the container is allowed to expand below the fold, and you are actually scrolling the rest of the container into view and not only the content.

The prod version allows for scrolling from below the fold into view because there is a wrapper around the container which is not present in the current usage of the radix component. I would have to create this wrapper manually and wrap `DialogPrimitive.Content`. This would also need to update where the styling is assigned to here.

I prefer the UX of prod, but wanted to show you what my quick fix looks like. If you are in agreement to maintain what is in prod, should I address it here or in a separate PR? I do not see this as a big fix, but I don't want to bloat or extend the life of this PR any more than necessary.

Update: I saw in Chromatic that my approach resulted in a side-effect of expanding the height without a max. (you'll see this when you review) This is correct behavior, but should not be employed unless there is a max-height that should be defined from the design. I think this further justifies addressing how to approach adding the content wrapper I described above.

Reply:
Looking at this, I think this effect will be easier if we make the Overlay wrap the Content (right now it is a sibling which make it hard to place it).

Following [this guide](https://www.radix-ui.com/primitives/docs/components/dialog#scrollable-overlay), we could do

```tsx
// this
const DialogContent = () => {
  return (
    <DialogPortal>
      <DialogOverlay>
        <DialogPrimitive.Content>
          {children}
        </DialogPrimitive.Content>
      </DialogOverlay>
    </DialogPortal>
  )
})

// instead of
<DialogPortal>
  <DialogOverlay />
  <DialogPrimitive.Content />
</DialogPortal>
```

and then,

```css
overlay => fixed inset-0 overflow-y-auto p-4
content => free, no positioning
```

Of course, this would be only for mobile screens, the previous styles should still apply to desktop resolutions.

Reply:
Ah I missed that. No problem!